sematic/api/endpoints/metrics.py

```diff
@@ -1,21 +1,145 @@
+    GroupBy,
+    MetricsFilter,
+    NoMetricError,
+    RollUp,
+    get_metrics_storage_plugins,
+)
 from sematic.plugins.metrics_storage.sql.sql_metrics_storage import SQLMetricsStorage
 
 logger = logging.getLogger(__name__)
 
 
+def _get_rollup(args: Dict[str, str]) -> RollUp:
+    rollup_arg = flask.request.args.get("rollup")
+
+    if rollup_arg is None:
+        return rollup_arg
+
+    if rollup_arg.isdigit():
+        return int(rollup_arg)
+
+    if rollup_arg == "auto":
+        return cast(Literal["auto"], rollup_arg)
+
+    raise ValueError("Incorrect value for rollup")
+
+
+@sematic_api.route("/api/v1/metrics/<metric_name>", methods=["GET"])
+@authenticate
+def get_metric_endpoint(user: Optional[User], metric_name: str) -> flask.Response:
+    metrics_storage = SQLMetricsStorage()
+
+    from_time_ts = float(flask.request.args.get("from_time", 0))
+    to_time_ts = float(flask.request.args.get("to_time", time.time()))
+
+    try:
+        labels: Dict = json.loads(flask.request.args.get("labels", "{}"))
+    except Exception as e:
+        return jsonify_error(
+            f"Unable to deserialize labels: {e}", HTTPStatus.BAD_REQUEST
+        )
+
+    try:
+        rollup = _get_rollup(flask.request.args)
+    except ValueError:
+        return jsonify_error(
+            "Incorrect value for rollup. Expected an integer or auto.",
+            HTTPStatus.BAD_REQUEST,
+        )
+
+    filter = MetricsFilter(
+        name=metric_name,
+        from_time=datetime.datetime.fromtimestamp(from_time_ts),
+        to_time=datetime.datetime.fromtimestamp(to_time_ts),
+        labels=labels,
+    )
+
+    try:
+        group_bys = [
+            GroupBy(value)
+            for value in flask.request.args.get("group_by", "").split(",")
+            if len(value) > 0
+        ]
+    except ValueError:
+        options = ", ".join([gb.name for gb in GroupBy])
+        return jsonify_error(
+            f"Unrecognized group_by. Options are {options}", HTTPStatus.BAD_REQUEST
+        )
+
+    try:
+        series = metrics_storage.get_aggregated_metrics(filter, group_bys, rollup)
+    except NoMetricError as e:
+        return jsonify_error(str(e), HTTPStatus.NOT_FOUND)
+
+    return flask.jsonify(content=series)
+
+
+@sematic_api.route("/api/v1/metrics", methods=["POST"])
+@authenticate
+def log_metric_endpoint(user: Optional[User]) -> flask.Response:
+    plugin_class = get_metrics_storage_plugins(default=[SQLMetricsStorage])[0]
+
+    plugin = plugin_class()
+
+    payload: Dict[str, Any] = flask.request.json  # type: ignore
+
+    metric_points = []
+
+    for metric_point_ in payload.get("metric_points", []):
+        metric_point = MetricPoint.from_json_encodable(metric_point_)
+        if user is not None:
+            metric_point.labels["user_id"] = user.id
+
+        metric_points.append(metric_point)
+
+    plugin.store_metrics(metric_points)
+
+    # broadcast_metrics_update(metric_points, user)
```

Comment:
Will this be enabled in a future Dashboard PR?

Reply:
I discovered doing this work that when the server runs locally, it is unable to query itself (in this case `POST /api/v1/events`). This works fine when deployed in K8s.
This never came up because when running the server locally and executing pipelines with `LocalResolver`, all requests to `/api/v1/events` are done from the resolver, not from other API endpoints.
I don't have the failure mode at hand, but we should fix this separately.